src/libcsg/modules/io/pdbreader.cc

```diff
@@ -27,25 +29,35 @@ using namespace std;
 
 bool PDBReader::ReadTopology(string file,  Topology &top)
 {
-   _topology = true;
-   top.Cleanup();
+    _topology = true;
+    top.Cleanup();
 
-   _fl.open(file.c_str());
+    _fl.open(file.c_str());
     if(!_fl.is_open())
         throw std::ios_base::failure("Error on open topology file: " + file);
 
-   NextFrame(top);
-
+    NextFrame(top);
     _fl.close();
 
     return true;
 }
 
-bool PDBReader::Open(const string &file)
-{
+bool PDBReader::Open(const string &file){
+
+    boost::filesystem::path filepath(file.c_str());
+
+    // Check that the file has the correct extension
+    if(boost::filesystem::extension(filepath).size()==0 ){
+        throw std::ios_base::failure("ERROR on opening .pdb file'"
+                +file+"' - extension is expected, use .pdb");
+    }else if(boost::filesystem::extension(filepath)!=".pdb"){
+        throw std::ios_base::failure("Error on opening .pdb file '"
+                +file+"' - wrong extension, use .pdb");
+    }
```

Comment:
Why is that needed? The trj/topo reader factory already make sure that this code is only called for a pdb files. Also if it wouldn't be called pdb (but e.g. pdbx) that would be fine, too.

Reply:
It was in one of the other file readers which I used as a template. I can remove it.

Reply:
The dlpolyreader can handle two extensions, plus it actually opens files with hard-codes names depending on that extension (yes, that is how dlpoly works!), so don't worry about that for the pdbreader and drop that block!